src/data/list/basic.lean

```diff
@@ -2093,6 +2093,14 @@ end
 | b []      l₂ := rfl
 | b (a::l₁) l₂ := by simp only [cons_append, foldr_cons, foldr_append b l₁ l₂]
 
+theorem foldl_fixed {a : α} : Π l : list β, foldl (λ a b, a) a l = a
```

Comment:
If you really wanted, you could add a variant of this for
```lean
theorem foldl_combinator_K {a : α} : Π l : list β, foldl combinator.K a l = a
```
but I doubt it would be used

Reply:
Well, it can't possibly hurt.